pythonx/ncm2_webcomplete.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+import re
+import subprocess
+from os.path import dirname, abspath, join, pardir, expandvars, expanduser
+from subprocess import check_output, PIPE
+from threading import Thread, Event
+
+import vim
+from ncm2 import Ncm2Source, getLogger
+
+
+logger = getLogger(__name__)
+
+
+def log(msg):
+    from datetime import datetime
+    timestamp = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S,%f")
+    with open('/tmp/ncm2-webcomplete.log', 'a') as file_:
+        file_.write('%s %s\n' % (timestamp, msg))
+
+
+class Source(Ncm2Source):
+    def __init__(self, nvim):
+        super(Source, self).__init__(nvim)
+
+        filedir = dirname(abspath(__file__))
+        projectdir = abspath(join(filedir, pardir))
+        self.__script = join(projectdir, 'sh', 'webcomplete')
+        self.__script = self.nvim.eval('g:ncm2_webcomplete_script') \
```

Comment:
This line of code throws an error.

Even though you have a fallback for the case the variable `g:ncm2_webcomplete_script`, it still throws an error when it doesn't exist.

```
[ncm2_webcomplete@yarp] Traceback (most recent call last):
[ncm2_webcomplete@yarp]   File "/Users/thalesmello/.config/nvim/plugged/nvim-yarp/pythonx/yarp.py", line 59, in <module>
[ncm2_webcomplete@yarp]     module_obj = importlib.import_module(module)
[ncm2_webcomplete@yarp]   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
[ncm2_webcomplete@yarp]     return _bootstrap._gcd_import(name[level:], package, level)
[ncm2_webcomplete@yarp]   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
[ncm2_webcomplete@yarp]   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
[ncm2_webcomplete@yarp]   File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
[ncm2_webcomplete@yarp]   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
[ncm2_webcomplete@yarp]   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
[ncm2_webcomplete@yarp]   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
[ncm2_webcomplete@yarp]   File "/Users/thalesmello/.config/nvim/plugged/webcomplete.vim/pythonx/ncm2_webcomplete.py", line 49, in <module>
[ncm2_webcomplete@yarp]     source = Source(vim)
[ncm2_webcomplete@yarp]   File "/Users/thalesmello/.config/nvim/plugged/webcomplete.vim/pythonx/ncm2_webcomplete.py", line 30, in __init__
[ncm2_webcomplete@yarp]     self.__script = self.nvim.eval('g:ncm2_webcomplete_script') \
[ncm2_webcomplete@yarp]   File "/usr/local/lib/python3.7/site-packages/neovim/api/nvim.py", line 280, in eval
[ncm2_webcomplete@yarp]     return self.request('nvim_eval', string, **kwargs)
[ncm2_webcomplete@yarp]   File "/usr/local/lib/python3.7/site-packages/neovim/api/nvim.py", line 170, in request
[ncm2_webcomplete@yarp]     res = self._session.request(name, *args, **kwargs)
[ncm2_webcomplete@yarp]   File "/usr/local/lib/python3.7/site-packages/neovim/msgpack_rpc/session.py", line 100, in request
[ncm2_webcomplete@yarp]     raise self.error_wrapper(err)
[ncm2_webcomplete@yarp] neovim.api.nvim.NvimError: b'Vim:E121: Undefined variable: g:ncm2_webcomplete_script'
[ncm2_webcomplete@yarp] Job is dead. cmd=['python3', '-u', '/Users/thalesmello/.config/nvim/plugged/nvim-yarp/pythonx/yarp.py', '/var/folders/4f/4r6phsjj0jn19qywlx5zxrjc0000gn/T/nvimDey8pj/0', 8, 'ncm2_webcomplete']
```

An alternative might be `self.__script = self.self.nvim.eval("get(g:,'ncm2_webcomplete_script', '')" or self.__script`